android/app/src/main/java/app/organicmaps/util/Utils.java

```diff
@@ -166,6 +166,37 @@ public static boolean isIntentSupported(@NonNull Context context, @NonNull Inten
     return null;
   }
 
+  @SuppressLint("BatteryLife")
+  @RequiresApi(api = Build.VERSION_CODES.M)
+  public static @Nullable Intent makeAppBatteryOptimizationIntent(@NonNull Context context)
+  {
+    Intent intent = new Intent();
+    intent.setData(Uri.parse("package:" + context.getPackageName()));
+    intent.setAction(android.provider.Settings.ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS);
+    if (isIntentSupported(context, intent))
+      return intent;
+    intent.setAction(android.provider.Settings.ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS);
```

Comment:
> What is the difference between ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS 

In documentation it says this will try to turn off battery optimisation by itself without requesting the user.

>ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS?

This one asks permission.